_posts/2019-05-24-varying-slopes/varying_slopes.Rmd

```diff
@@ -0,0 +1,748 @@
+
+Before we code the model, let's quickly get library loading out of the way. Importantly, again just like in the previous post, we need to install a `master` build of TensorFlow Probability, as we're making use of very new features not yet available in the current release version. The same goes for the R packages `tensorflow` and `tfprobability`: Please install the respective development versions from github. 
+
+```{r}
+devtools::install_github("rstudio/tensorflow")
+devtools::install_github("rstudio/tfprobability")
+
+# this will install the latest nightlies of TensorFlow as well as TensorFlow Probability
+tensorflow::install_tensorflow(version = "nightly")
+
+library(tensorflow)
+tf$compat$v1$enable_v2_behavior()
+
+library(tfprobability)
+
+library(tidyverse)
+library(zeallot)
+library(abind)
+library(gridExtra)
+library(HDInterval)
+library(ellipse)
+```
+
+
+Now here is the model definition. We'll go through it step by step in an instant.
+
+
+```{r}
+model <- function(cafe_id) {
+  tfd_joint_distribution_sequential(
+      list(
+        # rho, the prior for the correlation matrix between intercepts and slopes
+        tfd_lkj(2, 2, input_output_cholesky = TRUE), 
+        # sigma, prior variance for the waiting time
+        tfd_sample_distribution(tfd_exponential(rate = 1), sample_shape = 1),
+        # sigma_cafe, prior of variances for intercepts and slopes (vector of 2)
+        tfd_sample_distribution(tfd_exponential(rate = 1), sample_shape = 2), 
+        # b, the prior mean for the slopes
+        tfd_sample_distribution(tfd_normal(loc = -1, scale = 0.5), sample_shape = 1),
+        # a, the prior mean for the intercepts
+        tfd_sample_distribution(tfd_normal(loc = 5, scale = 2), sample_shape = 1), 
+        # mvn, multivariate distribution of intercepts and slopes
+        # shape: batch size, 20, 2
+        function(a,b,sigma_cafe,sigma,chol_rho) 
+          tfd_sample_distribution(
+            tfd_multivariate_normal_tri_l(
+              loc = tf$concat(list(a,b), axis = -1L),
+              scale_tril = tf$linalg$LinearOperatorDiag(sigma_cafe)$matmul(chol_rho)),
+            sample_shape = n_cafes),
+        # waiting time
+        # shape should be batch size, 200
+        function(mvn, a, b, sigma_cafe, sigma)
+          tfd_independent(
+            # need to pull out the correct cafe_id in the middle column
+            tfd_normal(
+              loc = (tf$gather(mvn[ , , 1], cafe_id, axis = -1L) +
+                       tf$gather(mvn[ , , 2], cafe_id, axis = -1L) * afternoon), 
+              scale=sigma),  # Shape [batch,  1]
+        reinterpreted_batch_ndims=1
+        )
+    )
+  )
+}
+```
+
+
+The first five distributions are priors. First,
```

Comment:
I think the "First," here should go with the rest of the sentence, which can either be above or below the code chunk that follows.

Reply:
Splitting them works in other places where the sentence is a bit more evenly broken up.

Reply:
oh yeah I see